packages/apps/human-app/frontend/src/router/components/layout/protected/drawer-navigation.tsx

```diff
@@ -0,0 +1,130 @@
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import CssBaseline from '@mui/material/CssBaseline';
+import { Stack } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { t } from 'i18next';
+import type { Dispatch, SetStateAction } from 'react';
+import { HumanLogoNavbarIcon } from '@/shared/components/ui/icons';
+import { Button } from '@/shared/components/ui/button';
+import { useIsMobile } from '@/shared/hooks/use-is-mobile';
+import { useHandleMainNavIconClick } from '@/shared/hooks/use-handle-main-nav-icon-click';
+import { TopMenuItemsList } from './top-menu-items-list';
+import { BottomMenuItemsList } from './bottom-menu-items-list';
+
+const drawerWidth = 240;
+
+export interface DrawerItem {
+  label: string;
+  link?: string;
+  href?: string;
+  icon?: JSX.Element;
+  disabled?: boolean;
+  onClick?: () => void;
+}
+
+export type MenuItem = DrawerItem | JSX.Element;
+interface DrawerNavigationProps {
+  open: boolean;
+  setDrawerOpen: Dispatch<SetStateAction<boolean>>;
+  topMenuItems?: MenuItem[];
+  bottomMenuItems?: MenuItem[];
+  signOut: () => void;
+}
+
+export function DrawerNavigation({
+  open,
+  setDrawerOpen,
+  topMenuItems,
+  bottomMenuItems,
+  signOut,
+}: Readonly<DrawerNavigationProps>) {
+  const navigate = useNavigate();
+  const isMobile = useIsMobile();
+  const handleMainNavIconClick = useHandleMainNavIconClick();
+
+  const handleItemClick = ({ disabled, href, link, onClick }: DrawerItem) => {
+    if (isMobile) setDrawerOpen(false);
+
+    if (disabled) return;
```

Comment:
I suppose it should go as a first check + actions because we probably want to do nothing in case item is "disabled", right?